### PR TITLE
Fix ideep call in MultiAdd

### PR DESCRIPTION
**Pull request by toslunar**
This PR fixes the multi_add part of #5055.

### CODE REVIEW

**Comment by kmaehashi** [comment]
This issue has been introduced in #4679.

File: chainer/functions/math/basic_math.py [hunk 1]
```diff
@@ -191,12 +191,11 @@ def forward(self, xs):
...
         if len(xs) == 1:
             return xs
-        y = None
-        if intel64.should_use_ideep('>=auto'):
-            bxs = numpy.broadcast_arrays(*xs)
-            if intel64.inputs_all_ready(bxs):
-                y = intel64.ideep.multi_add(bxs)
-        if y is None:
+        if (intel64.should_use_ideep('>=auto')
+                and intel64.inputs_all_ready(*xs)
```
Comment (kmaehashi) [inline chainer/functions/math/basic_math.py]: `inputs_all_ready` expects list of arrays as list. Just pass `xs`.

**Comment by kmaehashi** [comment]
I ran benchmark ([functions.math.basic_math.AddFunc.time_forward](https://github.com/chainer/chainer-benchmark/pull/25)) and confirmed performance regression fixed.

### this PR (with `inputs_all_ready` fixed)

```
=========== ========= ========= ==========
--                      count             
----------- ------------------------------
  backend       2         4         16    
=========== ========= ========= ==========
    gpu       1.95ms    6.56ms   28.26ms  
    cpu      26.11ms   45.05ms   156.65ms 
 cpu-ideep   10.06ms   17.91ms   57.19ms  
=========== ========= ========= ==========
```

### current master (3be74f5)

```
=========== ========= ========= ==========
--                      count             
----------- ------------------------------
  backend       2         4         16    
=========== ========= ========= ==========
    gpu       1.94ms    5.69ms   28.29ms  
    cpu      26.16ms   44.88ms   156.41ms 
 cpu-ideep   10.61ms   54.74ms   210.57ms 
=========== ========= ========= ==========
```

### 5.0.0a1

```
=========== ========= ========= ==========
--                      count             
----------- ------------------------------
  backend       2         4         16    
=========== ========= ========= ==========
    gpu       1.94ms    5.77ms   28.26ms  
    cpu      26.12ms   44.88ms   157.21ms 
 cpu-ideep   10.04ms   17.55ms   55.83ms  
=========== ========= ========= ==========
```

File: chainer/functions/math/basic_math.py (hunk 1 above)
Comment (toslunar) [reply chainer/functions/math/basic_math.py]: Oops, my manual merge at reverting c5f9195 is wrong.  Instead e1059f6 should be reverted too.

**Comment by toslunar** [comment]
Jenkins, test this please.

File: chainer/functions/math/basic_math.py (hunk 1 above)
Comment (toslunar) [reply chainer/functions/math/basic_math.py]: Fixed!

**Comment by mingxiaoh** [comment]
basic test showed that problem will be fixed by this commit.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**